### PR TITLE
#Enumerable

### PR DESCRIPTION
**Pull request by abrormukimov**
## Assignment #2. Enumerable

In this project, I used
- Methods;
- Blocks and Procs;
- Yield;
- Conditionals;
- Loops

### CODE REVIEW

**Comment by pelzolga123** [comment]
## Hi Abror!  
### You did a good job so far :+1: 
Still, there are changes required before this project can be approved:
* [ ] Stickler is not checking this Pull request, make sure that you set up everything correctly.
### Methods:
* #### My_each
* [ ] Should return an Enumerator if no block is given. For example:
`[1, 2, 3].my_each` - should return` #<Enumerator: [1, 2, 3]:my_each>`

*  #### My_each_with_index
* [ ] Should return an Enumerator if no block is given. For example:
`[1, 2, 3].my_each_with_index` - should return` #<Enumerator: [1, 2, 3]:my_each_with_index>`

*  #### My_select
* [ ] Should return an Enumerator if no block is given. For example:
`[1, 2, 3].my_select` - should return` #<Enumerator: [1, 2, 3]:my_select>`

*  #### My_all?
* [ ] Should return true if the block never returns false or nil. For example:
`[1, 2, 3].my_all?{|n| n <= 3}` should return true

* [ ] When no block or argument is given should return true if at least one of the collection is not false or nil. For example:
`[1, false, 'hi', []].my_all?` - should return false

* [ ]  When a class is passed as an argument should return true if all of the collection is a member of such class. For example:
`[1, 'dog', []].my_all?(Integer)` - should return false

* [ ] When a pattern other than Regex or a Class is given should return true if all of the collection matches the pattern. For example:
`['dog','door','dish'].my_all?(/d/)` - should return true

* [ ] When a Regex is passed as an argument should return true if all of the collection matches the pattern. For example:
`[3, 3, 3].my_all?(3)` - should return true

*  #### My_any?
* [ ] When no block or argument is given should return true if at least one of the collection is not false or nil. For example:
`[nil, false, nil, false].my_any?` - should return false

* [ ] When a class is passed as an argument should return true if at least one of the collection is a member of such class. For example:
`[1, 2, 3, 4, 5].my_any?(Integer)` - should return true

* [ ]  When a Regex is passed as an argument should return false if none of the collection matches the Regex. For example:
`['dog','door','dish'].my_any?(/z/)` - should return false

* [ ] When a pattern other than Regex or a Class is given should return false if none of the collection matches the pattern. For example:
`['dog', 'bird', 'fish'].my_any?('dog')` - should return true

*  #### My_none?
* [ ] When no block or argument is given should return true only if none of the collection members is true. For example:
`[nil, false, nil, false].my_none?` - should return true

* [ ] When a class is passed as an argument should return true if none of the collection is a member of such class. For example:
[1, 2 ,3].my_none?(String) - should return false

* [ ]  When a Regex is passed as an argument should return true only if none of the collection matches the Regex. For example:
`['dog','door','dish'].my_none?(/z/)` - should return true

* [ ] When a pattern other than Regex or a Class is given should return true only if none of the collection matches the pattern. For example:
`['dog', 'bird', 'fish'].my_none?(5)` - should return true

*  #### My_count
* [ ] Should return the number of items in enum through enumeration. For example:
`[1, 2 ,3, 4, 5, 6, 7, 8].my_count` - should return 6

* [ ] Should count the number of items in enum that are equal to an item if an argument is given. For example:
`[2 ,3, 4, 3, 6, 3].my_count(3)` - should return 3 

*  #### My_map
* [ ] Should return an Enumerator if no block is given. For example: 
`[1, 2 ,3].my_map` - should return `#<Enumerator: [1, 2, 3]:my_map>`

* [ ] Should return a new array with the results of running block once for every element in enum. For example: 
`[8, 4, 3, 11, 5, 6].my_map{|n| n < 6}` - should return `[false, true, true, false, true, false]`

*  #### My_inject
* [ ] Should combine all elements of enum by applying a binary operation, specified by a block. For example: 
`(1..7).my_inject(4){|prod, n| prod*n}` - should return 20160
 
* [ ] When a symbol is specified should combine each element of the collection by applying the symbol as a named method. For example: 
`[1, 2, 3, 4, 5].my_inject(:+)` - should return 15
### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: script.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# puts
+# puts "-----------"
+# [1, 2, 3, 4, 5].my_map { |x| puts x * 2 }
+# puts "-----------"
+# print [1, 2, 3, 4, 5, 6, 7].my_inject { |accum, num| accum + num }
+# puts
+# puts "-----------"
+# print [1, 2, 3, 5, 7].multiply_els { |accum, num| accum * num }
+# puts
+# puts "-----------"
```
Comment (pelzolga123) [inline script.rb:102]: Comments: It’s better to delete the commented code.

**Comment by abrormukimov** [comment]
On lines 34, 45, 56, 68 I used === sign. But rubocop gave me the error. Without it, code doesn't run correctly. I removed it so that I won't have the issue with the rubocop.

**Comment by pelzolga123** [comment]
## Amazing work :+1:
### Only one more method is not working correctly:

*  #### My_inject
* [ ] Should combine all elements of enum by applying a binary operation, specified by a block. For example: 
`(1..7).my_inject(4){|prod, n| prod*n}` - should return 20160
 
* [ ] When a symbol is specified should combine each element of the collection by applying the symbol as a named method. For example: 
`[1, 2, 3, 4, 5].my_inject(:+)` - should return 15

* [ ] Return `===` sign on 34, 45, 56, 68 lines. To make rubocop work, you can have `Style/CaseEquality` cop disabled as described [here](https://rubydoc.info/gems/rubocop/0.24.1/frames#Disabling_Cops_within_Source_Code) ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/style_case_equality.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421589/))
### Please kindly fix the above issues and send another code review request.


If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Pull request merged**